packages/core/src/dispatcher.ts

```diff
@@ -302,21 +311,16 @@ export class Dispatcher {
    * @param cid - Commit CID
    * @private
    */
-  async _restrictCommitSize(cid: CID | string): Promise<void> {
+  async _restrictCommitSize(cid: CID | string): Promise<number> {
     const asCid = typeof cid === 'string' ? CID.parse(cid) : cid
-    const stat = await this._shutdownSignal.abortable((signal) => {
+    const { size } = await this._shutdownSignal.abortable((signal) => {
       return this._ipfs.block.stat(asCid, {
```

Comment:
Shy still do this one remotely?

Reply:
I believe we should. `_restrictCommitSize` is adjacent to [`dag.get`](https://github.com/ceramicnetwork/js-ceramic/pull/2820/files/3cd767b54de60bded408dc69cb59944006a8f17e#diff-e522280df0e1c4c8fe4f157393e23872535ee7bab6d33a2930c26b18f7411d19R209) call, which returns decoded IPLD structure. To check that w/o a call to IPFS we'd need to convert the structure back to block (useless) or decode the block on js-ceramic side (thus dealing with all of the codecs available on our own). A choice between scylla and charybdis IMO, which we better fly over.

To be honest, I am not yet sure about the extent of the performance improvement this PR brings. Measuring it now.

Reply:
If we want to minimize load on IPFS during high traffic this seems worth it? One call to `block.get` instead of two calls (`block.stat` and `dag.get`).
I assume we have the codecs in the js code anyway?